server/py/services/api/tests/unit/runtime_handlers/test_kubejob.py

```diff
@@ -972,6 +977,33 @@ async def test_retry_pod_deleted_before_first_attempt(
             },
         )
 
+    @pytest.mark.parametrize(
+        "pod_retry_label, run_retry_count, expected_result",
+        [
+            # first run, no retry label means pod is valid and not outdated
+            (None, 0, False),
+            # retry count > 0 and no retry label is present, pod is outdated
+            (None, 1, True),
+            # pod attempt is older than current run retry, pod is outdated
+            ("1", 2, True),
+            # pod attempt equals current run retry, pod is still valid
+            ("2", 2, False),
+        ],
+    )
```

Comment:
Is an edge case possible where `pod_retry_label > run_retry_count`?
If so - what do we do in this case?